python/pyspark/sql/connect/dataframe.py

```diff
@@ -323,6 +323,14 @@ def unionByName(self, other: "DataFrame", allowMissingColumns: bool = False) ->
     def where(self, condition: Expression) -> "DataFrame":
         return self.filter(condition)
 
+    def summary(self, *statistics: str) -> "DataFrame":
+        _statistics: List[str] = list(statistics)
+        assert all(isinstance(s, str) for s in _statistics)
```

Comment:
Given `def summary(self, *statistics: str) -> "DataFrame":` is public API so there could be some mis-use passing into non-str parameters for `statistics`, is this common practice to just assert without giving a message?

Reply:
In contrast, I guess the assert in `plan.py` is ok because that is internal API and we can implement it right
```
    def __init__(self, child: Optional["LogicalPlan"], function: str, **kwargs: Any) -> None:
        super().__init__(child)
        assert function in ["summary"]
        self.function = function
  ```

Reply:
agree, should give the error message, will update